### PR TITLE
update migration dtos

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Updated migration dtos so they work with the result we get from HORNET

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

```Rust
    let migration_payload = r#"{
           "type":3,
           "migratedAt":42933,
           "funds":[
              {
                 "tailTransactionHash":"e0ce1933a900d02e0a43e2ba4e922d1a706a36a117ce1eb9ad0a74d39ee722c41b00cd227232efd76f5997e91bf8000000",
                 "address":{
                    "type":0,
                    "address":"18d65c3b901af94bb905eac6a74799134e15dda07c198ed54bb76d23992116f6"
                 },
                 "deposit":55590605665
              }
           ],
           "transaction":{
              "type":4,
              "input":{
                 "type":1,
                 "milestoneId":"95262b0bd51a53bd8a6e20df0e5fc4a0ae342783e8753f0c1a3bf57a4376e271"
              },
              "output":{
                 "type":2,
                 "amount":2778797567001116
              }
           },
           "final":true
        }"#;

     let r: ReceiptPayloadDto = serde_json::from_str(&migration_payload).unwrap();
```

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-ledger/src/types/receipt.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ impl Receipt {
             migrated_amount =
                 migrated_amount
                     .checked_add(funds.output().amount())
-                    .ok_or(Error::InvalidMigratedFundsAmount(
+                    .ok_or_else(|| Error::InvalidMigratedFundsAmount(
```
Comment (thibault-martinez) [inline bee-ledger/src/types/receipt.rs:44]: Why ?

File: bee-ledger/src/types/receipt.rs [hunk 2]
```diff
@@ -59,7 +59,7 @@ impl Receipt {
             .inner()
             .amount()
             .checked_sub(migrated_amount)
-            .ok_or(Error::InvalidMigratedFundsAmount(
+            .ok_or_else(|| Error::InvalidMigratedFundsAmount(
```
Comment (thibault-martinez) [inline bee-ledger/src/types/receipt.rs:62]: Why ?

File: bee-api/bee-rest-api/src/types/dtos.rs [hunk 3]
```diff
@@ -863,9 +866,9 @@ impl TryFrom<&MigratedFundsEntry> for MigratedFundsEntryDto {
 
     fn try_from(value: &MigratedFundsEntry) -> Result<Self, Self::Error> {
         Ok(MigratedFundsEntryDto {
-            tail_transaction_hash: value.tail_transaction_hash().as_ref().into(),
+            tail_transaction_hash: hex::encode(value.tail_transaction_hash().as_ref()),
```
Comment (thibault-martinez) [inline bee-api/bee-rest-api/src/types/dtos.rs:869]: Double-checking: is this supposed to be hex ? Or a tryte string as this is from the legacy network ?

File: bee-api/bee-rest-api/src/types/dtos.rs [hunk 4]
```diff
@@ -877,14 +880,13 @@ impl TryFrom<&MigratedFundsEntryDto> for MigratedFundsEntry {
     fn try_from(value: &MigratedFundsEntryDto) -> Result<Self, Self::Error> {
         let entry = MigratedFundsEntry::new(
             TailTransactionHash::new(
-                value
-                    .tail_transaction_hash
-                    .as_ref()
+                hex::decode(value.tail_transaction_hash.clone())
```
Comment (thibault-martinez) [inline bee-api/bee-rest-api/src/types/dtos.rs:883]: Same question

File: bee-ledger/src/types/receipt.rs (hunk 1 above)
Comment (Thoralf-M) [reply bee-ledger/src/types/receipt.rs:44]: There was a clippy warning, should I undo it?

File: bee-api/bee-rest-api/src/types/dtos.rs (hunk 3 above)
Comment (Thoralf-M) [reply bee-api/bee-rest-api/src/types/dtos.rs:869]: Not sure, I just know that we get `e0ce1933a900d02e0a43e2ba4e922d1a706a36a117ce1eb9ad0a74d39ee722c41b00cd227232efd76f5997e91bf8000000` from HORNET

File: bee-ledger/src/types/receipt.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-ledger/src/types/receipt.rs:44]: Ah alright, weird. No no don't undo, thanks.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**